compiler/rustc_parse/src/parser/attr_wrapper.rs

```diff
@@ -339,17 +339,17 @@ impl<'a> Parser<'a> {
         let parser_replacements_end = self.capture_state.parser_replacements.len();
 
         assert!(
-            !(self.break_last_token && matches!(capture_trailing, Trailing::Yes)),
-            "Cannot set break_last_token and have trailing token"
+            !(self.break_last_token > 0 && matches!(capture_trailing, Trailing::Yes)),
+            "Cannot have break_last_token > 0 and have trailing token"
         );
 
         let end_pos = self.num_bump_calls
             + capture_trailing as u32
-            // If we 'broke' the last token (e.g. breaking a '>>' token to two '>' tokens), then
+            // If we "broke" the last token (e.g. breaking a `>>` token to two `>` tokens), then
             // extend the range of captured tokens to include it, since the parser was not actually
             // bumped past it. When the `LazyAttrTokenStream` gets converted into an
             // `AttrTokenStream`, we will create the proper token.
-            + self.break_last_token as u32;
+            + if self.break_last_token == 0 { 0 } else { 1 };
```

Comment:
Need an explanation why `if self.break_last_token == 0 { 0 } else { 1 }` and not `self.break_last_token`.